engine/src/test/java/io/camunda/zeebe/engine/state/distribution/DistributionStateTest.java

```diff
@@ -165,10 +170,107 @@ public void shouldFailToAddPendingDistributionIfNoCommandDistributionExists() {
             "Foreign key DbLong{1} does not exist in COMMAND_DISTRIBUTION_RECORD");
   }
 
+  @Test
+  public void shouldIterateOverPendingDistributions() {
+    // given
+    final var distributionKey = 1L;
+    final var pendingDistribution = createCommandDistributionRecord();
+    distributionState.addCommandDistribution(distributionKey, pendingDistribution);
+
+    final int partitionId3 = 3;
+    final int partitionId2 = 2;
+    distributionState.addPendingDistribution(distributionKey, partitionId2);
+    distributionState.addPendingDistribution(distributionKey, partitionId3);
+
+    // when
+    final List<VisitedPendingDistribution> visits = new ArrayList<>();
+    distributionState.foreachPendingDistribution(
+        (key, commandDistributionRecord) ->
+            visits.add(new VisitedPendingDistribution(key, commandDistributionRecord)));
+
+    // then
+    assertThat(visits)
+        .allSatisfy(visited -> assertThat(visited.key()).isEqualTo(distributionKey))
+        .allSatisfy(
+            visited ->
+                Assertions.assertThat(visited.record())
+                    .hasIntent(pendingDistribution.getIntent())
+                    .hasValueType(pendingDistribution.getValueType())
+                    .hasCommandValue(pendingDistribution.getCommandValue()))
+        .extracting(VisitedPendingDistribution::record)
+        .extracting(CommandDistributionRecord::getPartitionId)
+        .describedAs("Expect that pending distributions are visited for all other partitions")
+        .containsExactly(partitionId2, partitionId3);
+  }
+
+  @Test
+  public void shouldIterateOverMultiplePendingDeployments() {
+    // given
+    final var distributions = new Long2ObjectHashMap<CommandDistributionRecord>();
+    final int partitionId2 = 2;
+    final int partitionId3 = 3;
+    for (int distributionKey = 1; distributionKey <= 5; distributionKey++) {
+      final var pendingDistribution = createCommandDistributionRecord();
+      distributions.put(distributionKey, pendingDistribution);
+      distributionState.addCommandDistribution(distributionKey, pendingDistribution);
+
+      distributionState.addPendingDistribution(distributionKey, partitionId2);
+      distributionState.addPendingDistribution(distributionKey, partitionId3);
```

Comment:
🔧 We have to do these calls in all these test cases. Might be nice to extract it to a separate method.